transaction/std/src/transaction_builder.rs

```diff
@@ -91,6 +181,10 @@ impl<FPR: FogPubkeyResolver> TransactionBuilder<FPR> {
     /// Caution: This method should not be used without fully understanding the
     /// implications.
     ///
+    /// Deprecation note: This method will not be public in future versions of
+    /// this crate, we believe the only legitimate use of this is now served
+    /// by add_change_output.
+    ///
```

Comment:
Reviewers: please let me know if you agree with this

Reply:
Whats currently relying on it being public? I think only libmobilecoin bindings?
Maybe we can deprecate it immediately if the Swift counterpart does not rely on it.

Reply:
I think maybe swift is the only thing that uses it, and i think `add_change_output` is going to be the new thing

maybe we can just mark this non-pub right now, but i dont think i can make a commit that will keep the swift bindings happy. i could put `unimplemented!()` in the body of the call i guess or something like this

Reply:
if it doesn't need to happen in this PR i'd rather land this PR so people can start integrating, and then i'll deal with the deprecation of api after that